File: Chess.Game/Pieces/Knight.cs
using System;
using System.Collections.Generic;

namespace Chess.Game.Pieces
{
    public class Knight : IPiece
    {
        public bool Moved { get; set; }
        public Color Color { get; set; }

        public IList<Location> GetLegalMoves(Location currentLocation)
        {
            throw new NotImplementedException();
        }

        public bool IsMoveLegal(Location current, Location newLocation)
        {
            if (newLocation.CurrentColor == Color)
                return false;

            var fileChange = Math.Abs((int) newLocation.File - (int) current.File);
            var rankChange = Math.Abs(newLocation.Rank - current.Rank);

            return fileChange == 2 && rankChange == 1 ||
                   fileChange == 1 && rankChange == 2;
        }
    }
}

File: Chess.Web/ClientApp/pieces/Knight.ts
import { Location } from '../store/Game';
import Piece from './Piece';
import Color from './Color';

class Knight implements Piece {
    color: Color;
    moves: number;

    constructor(moves: number, color: Color) {
        this.moves = moves;
        this.color = color;
    }

    isMoveLegal(current: Location, newLocation: Location, board: Location[][]) {
        if (newLocation.piece && newLocation.piece.color == this.color)
            return false;

        const fileChange = Math.abs(newLocation.file - current.file);
        const rankChange = Math.abs(newLocation.rank - current.rank);

        return fileChange == 2 && rankChange == 1 ||
                fileChange == 1 && rankChange == 2;
    };

    getImageName() {
        const image = this.color == Color.White ? "white_knight" : "black_knight";

        return image;
    }
}

export default Knight;
